src/api/routes/awala.routes.ts

```diff
@@ -0,0 +1,96 @@
+import type { RouteOptions } from 'fastify';
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import type { JSONSchema, FromSchema } from 'json-schema-to-ts';
+
+import { HTTP_STATUS_CODES } from '../../utilities/http.js';
+import type { PluginDone } from '../../utilities/fastify/PluginDone.js';
+import type { FastifyTypedInstance } from '../../utilities/fastify/FastifyTypedInstance.js';
+import { MEMBER_BUNDLE_REQUEST_SCHEMA } from '../../schemas/awala.schema.js';
+import { AwalaProblemType } from '../../AwalaProblemType.js';
+import { createMemberBundleRequest } from '../../awala.js';
+import type { ServiceOptions } from '../../serviceTypes.js';
+
+const ajv = addFormats(new Ajv());
+
+type ValidationResult<Schema extends JSONSchema> = FromSchema<Schema> | string;
+
+function validateMessage<Schema extends JSONSchema>(
+  value: unknown,
+  schema: Schema,
+): ValidationResult<Schema> {
+  if (!ajv.validate(schema, value)) {
+    return ajv.errorsText(ajv.errors);
+  }
+
+  return value as ValidationResult<Schema>;
+}
+
+async function processMemberBundleRequest(
+  data: unknown,
+  options: ServiceOptions,
+): Promise<AwalaProblemType | undefined> {
```

Comment:
It doesn't look idiomatic to do use `Promise<undefined>` and its corresponding `return undefined`. Doesn't the following work?

```suggestion
): Promise<AwalaProblemType | void> {
```